agent/exploits/cve_2024_9487.py

```diff
@@ -0,0 +1,92 @@
+It affected versions before 3.15 and was fixed in 3.11.16, 3.12.10, 3.13.5, and 3.14.2. The vulnerability was reported through GitHub's Bug Bounty program."""
+RISK_RATING = "CRITICAL"
+FIXED_VERSIONS = {
+    "3.11": "3.11.6",
+    "3.12": "3.12.10",
+    "3.13": "3.13.5",
+    "3.14": "3.14.2",
+}
+
+MAX_FIXED_VERSION = "3.15"
+MAX_NONFIXED_VERSION = "3.10"
+
+
+@exploits_registry.register
+class CVE20249487Exploit(webexploit.WebExploit):
+    accept_request = definitions.Request(method="GET", path="/")
+    check_request = definitions.Request(method="GET", path="/")
+    accept_pattern = [re.compile("GitHub\sEnterprise\sServer\s\d+\.\d+\.\d+")]
+    version_pattern = re.compile("GitHub\sEnterprise\sServer\s(\d+\.\d+\.\d+)")
+
+    metadata = definitions.VulnerabilityMetadata(
+        title=VULNERABILITY_TITLE,
+        description=VULNERABILITY_DESCRIPTION,
+        reference=VULNERABILITY_REFERENCE,
+        risk_rating=RISK_RATING,
+    )
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        """Rule to detect specific vulnerability on a specific target.
+
+        Args:
+            target: Target to scan
+
+        Returns:
+            List of identified vulnerabilities.
+        """
+        session = requests.Session()
+        session.max_redirects = MAX_REDIRECTS
+        session.verify = False
+
+        vulnerabilities: list[definitions.Vulnerability] = []
+
+        target_endpoint = urlparse.urljoin(target.origin, self.check_request.path)
+
+        try:
+            req = requests.Request(
+                method=self.check_request.method,
+                url=target_endpoint,
+                data=self.check_request.data,
+            ).prepare()
+            resp = session.send(req, timeout=DEFAULT_TIMEOUT.seconds)
+        except requests_exceptions.RequestException:
+            return vulnerabilities
+
+        if (matched := self.version_pattern.findall(resp.text)) != []:
+            for extracted_version in matched:
+                if isinstance(extracted_version, tuple):
+                    extracted_version = extracted_version[0]
+                if version.parse(extracted_version) < version.parse(MAX_FIXED_VERSION):
+                    if version.parse(extracted_version) < version.parse(
+                        MAX_NONFIXED_VERSION
+                    ):
+                        vulnerability = self._create_vulnerability(target)
+                        vulnerabilities.append(vulnerability)
+                        continue
+                    version_prefix = ".".join(extracted_version.split(".")[:2])
+                    if version.parse(extracted_version) < version.parse(
+                        FIXED_VERSIONS[version_prefix]
+                    ):
+                        vulnerability = self._create_vulnerability(target)
+                        vulnerabilities.append(vulnerability)
```

Comment:
You are reporting multiple vulnurabilities, for multiple versions that are found in a single target!
This is not right, you should have only one version and one vulnerability.